main.tf

```diff
@@ -203,6 +218,46 @@ module "eks" {
 }
 
 
+resource "aws_secretsmanager_secret" "cloudquery_secret" {
+  name                    = var.name
+  recovery_window_in_days = 0
+}
+
+resource "aws_secretsmanager_secret_version" "cloudquery_secret_version" {
+  secret_id = aws_secretsmanager_secret.cloudquery_secret.id
+  secret_string = jsonencode({
+    "CQ_VAR_DSN" : "postgres://${module.rds.db_instance_username}:${module.rds.db_instance_password}@${module.rds.db_instance_endpoint}/${module.rds.db_instance_name}",
+  })
+}
+
+
+module "iam_policy" {
+  source  = "terraform-aws-modules/iam/aws//modules/iam-policy"
+  version = "~> 4"
+
+  name        = "${var.name}-secretsmanager"
+  path        = "/"
+  description = "Access to CloudQuery secrets in AWS secret manager"
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": ["secretsmanager:GetSecretValue", "secretsmanager:DescribeSecret"],
+      "Effect": "Allow",
+      "Resource": "${aws_secretsmanager_secret_version.cloudquery_secret_version.arn}"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy_attachment" "irsa" {
```

Comment:
This resource is already covered in - https://github.com/terraform-aws-modules/terraform-aws-iam/blob/master/modules/iam-role-for-service-accounts-eks/main.tf#L41-L46

See line 113 above